Not compatible with Rider 2023.2 EAP 6

`WorkspaceModel` seems to have been removed in EAP 6, causing the previewer to crash.
Opening this issue to let you know, I'm not expecting a new release for every EAP version, they're EAP after all :)

```
java.lang.NoClassDefFoundError: com/intellij/workspaceModel/ide/WorkspaceModel
	at me.fornever.avaloniarider.idea.editor.actions.RunnableAssemblySelectorAction.<init>(RunnableAssemblySelectorAction.kt:70)
	at me.fornever.avaloniarider.idea.editor.AvaloniaPreviewEditorBase.<init>(AvaloniaPreviewEditorBase.kt:117)
	at me.fornever.avaloniarider.idea.editor.AvaloniaPreviewEditorBase.<init>(AvaloniaPreviewEditorBase.kt:53)
	at me.fornever.avaloniarider.idea.editor.AvaloniaRemotePreviewEditor.<init>(AvaloniaRemotePreviewEditor.kt:11)
	at me.fornever.avaloniarider.idea.editor.AvaloniaPreviewerXamlEditorExtension.createEditor(AvaloniaPreviewerXamlEditorExtension.kt:22)
	at com.jetbrains.rider.xaml.splitEditor.XamlSplitEditorImpl$loadPreviewEditorThroughExtensions$1.invoke(XamlSplitEditorImpl.kt:166)
	at com.jetbrains.rider.xaml.splitEditor.XamlSplitEditorImpl$loadPreviewEditorThroughExtensions$1.invoke(XamlSplitEditorImpl.kt:163)
	at com.jetbrains.rider.xaml.MarkupPreviewHost$executeWithPreviewPlatform$1$1.invoke(MarkupPreviewHost.kt:19)
	at com.jetbrains.rider.xaml.MarkupPreviewHost$executeWithPreviewPlatform$1$1.invoke(MarkupPreviewHost.kt:18)
	at com.jetbrains.rd.framework.impl.RdTaskKt$startAndAdviseSuccess$1.invoke(RdTask.kt:24)
	at com.jetbrains.rd.framework.impl.RdTaskKt$startAndAdviseSuccess$1.invoke(RdTask.kt:23)
	at com.jetbrains.rd.util.reactive.Signal.fire(Signal.kt:32)
	at com.jetbrains.rd.util.reactive.WriteOnceProperty$WriteOnceSignal.fire(Property.kt:122)
	at com.jetbrains.rd.util.reactive.WriteOnceProperty.setIfEmpty(Property.kt:86)
	at com.jetbrains.rd.framework.impl.CallSiteWiredRdTask$onWireReceived$4.invoke(RdTask.kt:134)
	at com.jetbrains.rd.framework.impl.CallSiteWiredRdTask$onWireReceived$4.invoke(RdTask.kt:133)
	at com.jetbrains.rd.framework.MessageBroker$RdWireableDispatchHelper$doDispatch$2.invoke(MessageBroker.kt:122)
	at com.jetbrains.rd.framework.MessageBroker$RdWireableDispatchHelper$doDispatch$2.invoke(MessageBroker.kt:115)
	at com.jetbrains.rdclient.protocol.RdDispatcher.flushAll(RdDispatcher.kt:103)
	at com.jetbrains.rdclient.protocol.RdDispatcher.flushAllRunnable$lambda$0(RdDispatcher.kt:29)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:208)
	at com.intellij.openapi.application.TransactionGuardImpl.access$100(TransactionGuardImpl.java:21)
	at com.intellij.openapi.application.TransactionGuardImpl$1.run(TransactionGuardImpl.java:190)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:861)
	at com.intellij.openapi.application.impl.ApplicationImpl$4.run(ApplicationImpl.java:478)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:79)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:121)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:41)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:789)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:740)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:734)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:759)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.kt:685)
	at com.intellij.ide.IdeEventQueue._dispatchEvent$lambda$10(IdeEventQueue.kt:589)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWithoutImplicitRead(ApplicationImpl.java:1485)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.kt:589)
	at com.intellij.ide.IdeEventQueue.access$_dispatchEvent(IdeEventQueue.kt:67)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:369)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:368)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:787)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:368)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:363)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$1(IdeEventQueue.kt:992)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:105)
	at com.intellij.ide.IdeEventQueueKt.performActivity(IdeEventQueue.kt:992)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$7(IdeEventQueue.kt:363)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:861)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:405)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
Caused by: java.lang.ClassNotFoundException: com.intellij.workspaceModel.ide.WorkspaceModel PluginClassLoader(plugin=PluginDescriptor(name=AvaloniaRider, id=avalonia-rider, descriptorPath=plugin.xml, path=~\AppData\Roaming\JetBrains\Rider2023.2\plugins\avaloniarider, version=1.0.4, package=null, isBundled=false), packagePrefix=null, state=active)
	at com.intellij.ide.plugins.cl.PluginClassLoader.loadClass(PluginClassLoader.kt:156)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	... 57 more
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,13 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/) and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [1.0.5] - 2023-06-30
+### Fixed
+- Correct DPI is now passed to the previewer process from the start.
+
+### Changed
+- The plugin is now only compatible with Rider 2023.2 EAP 6 and later.
+
 ## [1.0.4] - 2023-05-17
 ### Changed
 - The plugin is now only compatible with Rider 2023.2 EAP 1 and later.
@@ -36,4 +43,5 @@ The initial plugin release. The plugin provides the following features for JetBr
 [1.0.2]: https://github.com/ForNeVeR/AvaloniaRider/compare/v1.0.1...v1.0.2
 [1.0.3]: https://github.com/ForNeVeR/AvaloniaRider/compare/v1.0.2...v1.0.3
 [1.0.4]: https://github.com/ForNeVeR/AvaloniaRider/compare/v1.0.3...v1.0.4
-[Unreleased]: https://github.com/ForNeVeR/AvaloniaRider/compare/v1.0.4...HEAD
+[1.0.5]: https://github.com/ForNeVeR/AvaloniaRider/compare/v1.0.4...v1.0.5
+[Unreleased]: https://github.com/ForNeVeR/AvaloniaRider/compare/v1.0.5...HEAD

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -7,14 +7,14 @@ buildscript {
 
     // https://search.maven.org/artifact/com.jetbrains.rd/rd-gen
     dependencies {
-        classpath("com.jetbrains.rd:rd-gen:2023.2.1")
+        classpath("com.jetbrains.rd:rd-gen:2023.2.2")
     }
 }
 
 plugins {
     id("me.filippov.gradle.jvm.wrapper") version "0.14.0"
     id("org.jetbrains.changelog") version "2.0.0"
-    id("org.jetbrains.intellij") version "1.13.3"
+    id("org.jetbrains.intellij") version "1.14.2"
     id("org.jetbrains.kotlin.jvm") version "1.8.10"
 }
 

--- a/gradle.properties
+++ b/gradle.properties
@@ -1,6 +1,6 @@
 kotlin.stdlib.default.dependency=false
 
-pluginVersionBase=1.0.4
+pluginVersionBase=1.0.5
 
 # https://www.jetbrains.com/intellij-repository/snapshots/
 riderSdkVersion=2023.2-EAP6-SNAPSHOT

--- a/gradle.properties
+++ b/gradle.properties
@@ -3,7 +3,7 @@ kotlin.stdlib.default.dependency=false
 pluginVersionBase=1.0.4
 
 # https://www.jetbrains.com/intellij-repository/snapshots/
-riderSdkVersion=2023.2-EAP1-SNAPSHOT
+riderSdkVersion=2023.2-EAP6-SNAPSHOT
 untilBuildVersion=233.*
 
 buildRelease=false

--- a/src/rider/main/kotlin/me/fornever/avaloniarider/idea/editor/actions/RunnableAssemblySelectorAction.kt
+++ b/src/rider/main/kotlin/me/fornever/avaloniarider/idea/editor/actions/RunnableAssemblySelectorAction.kt
@@ -17,10 +17,10 @@ import com.intellij.openapi.rd.util.launchOnUi
 import com.intellij.openapi.util.io.FileUtil
 import com.intellij.openapi.vfs.VfsUtil
 import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.platform.backend.workspace.WorkspaceModel
 import com.intellij.util.application
 import com.intellij.util.io.systemIndependentPath
 import com.intellij.util.messages.MessageBus
-import com.intellij.workspaceModel.ide.WorkspaceModel
 import com.intellij.workspaceModel.ide.toPath
 import com.jetbrains.rd.util.lifetime.Lifetime
 import com.jetbrains.rd.util.reactive.*
@@ -45,7 +45,6 @@ import java.nio.file.Paths
 import javax.swing.JComponent
 import kotlin.io.path.nameWithoutExtension
 
-@Suppress("UnstableApiUsage")
 class RunnableAssemblySelectorAction(
     private val lifetime: Lifetime,
     private val project: Project,

--- a/src/rider/main/kotlin/me/fornever/avaloniarider/previewer/AvaloniaPreviewerSessionController.kt
+++ b/src/rider/main/kotlin/me/fornever/avaloniarider/previewer/AvaloniaPreviewerSessionController.kt
@@ -9,10 +9,13 @@ import com.intellij.openapi.project.Project
 import com.intellij.openapi.rd.util.*
 import com.intellij.openapi.util.Computable
 import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.platform.backend.workspace.WorkspaceModel
 import com.intellij.ui.scale.JBUIScale
 import com.intellij.util.application
-import com.intellij.workspaceModel.ide.WorkspaceModel
-import com.jetbrains.rd.util.lifetime.*
+import com.jetbrains.rd.util.lifetime.Lifetime
+import com.jetbrains.rd.util.lifetime.LifetimeDefinition
+import com.jetbrains.rd.util.lifetime.SequentialLifetimes
+import com.jetbrains.rd.util.lifetime.isAlive
 import com.jetbrains.rd.util.reactive.*
 import com.jetbrains.rd.util.throttleLast
 import com.jetbrains.rider.build.BuildHost
@@ -47,7 +50,6 @@ import java.time.Duration
 /**
  * The sources on this class are thread-free. Make sure to schedule them onto the proper threads if necessary.
  */
-@Suppress("UnstableApiUsage") // for workspace model
 class AvaloniaPreviewerSessionController(
     private val project: Project,
     outerLifetime: Lifetime,
@@ -267,7 +269,6 @@ class AvaloniaPreviewerSessionController(
         )
 
         val socket = withSyncIOBackgroundContext(lifetime) {
-            @Suppress("BlockingMethodInNonBlockingContext")
             ServerSocket(0).apply {
                 lifetime.onTermination { close() }
             }

--- a/src/rider/main/kotlin/me/fornever/avaloniarider/rider/ProjectModelNodeEx.kt
+++ b/src/rider/main/kotlin/me/fornever/avaloniarider/rider/ProjectModelNodeEx.kt
@@ -5,8 +5,8 @@ import com.intellij.openapi.diagnostic.Logger
 import com.intellij.openapi.diagnostic.debug
 import com.intellij.openapi.project.Project
 import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.platform.backend.workspace.WorkspaceModel
 import com.intellij.util.application
-import com.intellij.workspaceModel.ide.WorkspaceModel
 import com.jetbrains.rd.util.lifetime.Lifetime
 import com.jetbrains.rd.util.reactive.adviseUntil
 import com.jetbrains.rider.model.RdProjectDescriptor
@@ -28,7 +28,6 @@ val ProjectModelEntity.projectRelativeVirtualPath: String
         return names.joinToString("/")
     }
 
-@Suppress("UnstableApiUsage")
 suspend fun VirtualFile.getProjectContainingFile(lifetime: Lifetime, project: Project): ProjectModelEntity {
     val logger = Logger.getInstance("me.fornever.avaloniarider.rider.ProjectModelNodeExKt")
     val workspaceModel = WorkspaceModel.getInstance(project)

--- a/src/test/kotlin/me/fornever/avaloniarider/test/cases/MsBuildParameterCollectorTests.kt
+++ b/src/test/kotlin/me/fornever/avaloniarider/test/cases/MsBuildParameterCollectorTests.kt
@@ -1,6 +1,6 @@
 package me.fornever.avaloniarider.test.cases
 
-import com.intellij.workspaceModel.ide.WorkspaceModel
+import com.intellij.platform.backend.workspace.WorkspaceModel
 import com.jetbrains.rider.model.runnableProjectsModel
 import com.jetbrains.rider.projectView.solution
 import com.jetbrains.rider.projectView.workspace.getProjectModelEntities
@@ -23,7 +23,6 @@ class MsBuildParameterCollectorTests : AvaloniaIntegrationTest() {
 
     override fun getSolutionDirectoryName() = "MSBuildParameters"
 
-    @Suppress("UnstableApiUsage")
     @Test
     fun testLaunchSettingsParametersCollection() {
         val collector = MsBuildParameterCollector.getInstance(project)

--- a/src/test/kotlin/me/fornever/avaloniarider/test/cases/RunnableAssemblySelectorActionTests.kt
+++ b/src/test/kotlin/me/fornever/avaloniarider/test/cases/RunnableAssemblySelectorActionTests.kt
@@ -3,8 +3,8 @@ package me.fornever.avaloniarider.test.cases
 import com.intellij.execution.RunManager
 import com.intellij.openapi.actionSystem.ActionPlaces
 import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.platform.backend.workspace.WorkspaceModel
 import com.intellij.util.io.systemIndependentPath
-import com.intellij.workspaceModel.ide.WorkspaceModel
 import com.jetbrains.rd.util.lifetime.LifetimeDefinition
 import com.jetbrains.rd.util.reactive.IOptPropertyView
 import com.jetbrains.rd.util.reactive.OptProperty
@@ -62,7 +62,7 @@ class RunnableAssemblySelectorActionTests : AvaloniaIntegrationTest() {
         return RunnableAssemblySelectorAction(
             testLifetime,
             project,
-            @Suppress("UnstableApiUsage") WorkspaceModel.getInstance(project),
+            WorkspaceModel.getInstance(project),
             project.messageBus,
             RunManager.getInstance(project),
             AvaloniaSettings.getInstance(project),

